python/plugins/processing/core/parameters.py

```diff
@@ -79,13 +80,25 @@ def setValue(self, obj):
         of parameter.
         """
         if obj is None:
-            if not self.optional:
+            if not self.optional and self.default is None:
                 return False
-            self.value = None
-            return True
+            elif self.default is None:
+                self.value = None
+                return True
+            else:
```

Comment:
You are right @m-kuhn. The current approach assumes that before calling the algorithm, all parameter values will be set using setValue (which is what actually happens)

so the initialization is assumed to be done by the user calling the algorithm

Any better idea? We could add a method to check the values of parameters, before running an algorithm, to ensure that no mandatory parameters have been left without explicitely assigning them a value